examples/finance/bs_ivbp.ipynb

```diff
@@ -0,0 +1,649 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Black-Scholes with Devito\n",
+    "\n",
+    "Black-Scholes is a model used to determine the theoretical value of a call or a put option in finance. A way that traders might use it is to buy an options priced under the Black-Scholes value and sell options priced higher than the calculated value. Larger firms might use the model to hedge, or attempt to eliminate risk, by buying and selling the underlying asset in just the right way. Here we solve the partial differential equation and calculate theoretical \"fair\" values for an option from one year to expiration to the expiration date.\n",
+    "\n",
+    "Symbol definitions:\n",
+    "\n",
+    "* $S(t)$ is the price of the underlying asset at time $t$\n",
+    "* $V(t,S)$ is the value of the option at time $t$ and stock price $S$\n",
+    "* $K$ is the strike price of the option\n",
+    "* $r$ is the annualized risk-free interest rate\n",
+    "* $\\sigma$ is the standard deviation of the stock's returns\n",
+    "* $t$ is the time, in years, to expiration."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "dt,tmax,nt; 0.0005 1.0 2001\n",
+      "shape;    (81,)\n",
+      "origin;   (60,)\n",
+      "spacing;  (1.0,)\n",
+      "extent;   80\n"
+     ]
+    }
+   ],
+   "source": [
+    "from devito import (Eq, Grid, TimeFunction, Operator, solve, Constant, \n",
+    "                    SpaceDimension, configuration, SubDomain)\n",
+    "\n",
+    "from mpl_toolkits.mplot3d import Axes3D\n",
+    "from mpl_toolkits.mplot3d.axis3d import Axis\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib as mpl\n",
+    "from matplotlib import cm\n",
+    "\n",
+    "from sympy.stats import Normal, cdf\n",
+    "import numpy as np\n",
+    "import time as timer\n",
+    "\n",
+    "%matplotlib inline\n",
+    "mpl.rc('font', size=14)\n",
+    "\n",
+    "configuration[\"log-level\"] = 'INFO'\n",
+    "\n",
+    "# Constants\n",
+    "K = 100.0\n",
+    "r = 0.12\n",
+    "sigma = 0.1\n",
+    "smin = 70\n",
+    "smax = 130\n",
+    "\n",
+    "# If you want to try some different problems, uncomment these lines\n",
```

Comment:
Would it be possible to add a short description/explanation re. the various choices of constants?

Reply:
Robert you might add a link to the various papers we looked at to reproduce results.

Reply:
Added an explanation. I added the papers I looked at that weren't behind paywalls (the 3 point approximation one I showed you is on research gate and I think the UNL math one is a better example anyway)